JavaScript and style for right rail

tarvos21 opened the pull request:
- JavaScript from docs.js
- Style modifications
- Dependency
- typo

tarvos21 committed:
Add h2 headers in 4 tabs, make demo container a basic segment

tarvos21 committed:
Add jquery.address.js, modify js reference in footer

tarvos21 committed:
Style for right rail

- set tab width to 827px on screen wider than 1200px
- set left and right margins of demo container
- set right rail closely next to tab in container
- enlarge margin-top of the first child of right rail
- enlarge padding-bottom of the last child of each tab to make scrolling right
- make margin-left of the segment/container 0 to align it to other container

tarvos21 committed:
JavaScript code for right rail

- Copy from the original docs.js, but remove unrelated parts temporarily.
- Main functions modified include createMenu, tryCreateMenu, scrollTo
- Handle edge cases of Tab and Transition examples in Modules tab

Problem remained here and also in the original site:
- if one visits the site for the first time, or refreshes with F5, the right rail work fine, but if he clicks and enters another tab, then he clicks the tab he looks before, the right rail may have some problems, for example, the active item will not change color.

tarvos21 committed:
typo, colon

xiaohanyu commented on content/static/stylesheets/default.sass:
```diff
@@ -130,6 +130,23 @@ img
+  margin-right: 387px!important
+  .tab
+    width: 827px
+    .example:last-child:not(.position)
```
This line seems too large.

Normal page:

![image](https://cloud.githubusercontent.com/assets/1164623/21141270/3914fc58-c177-11e6-87a8-c07984624fb5.png)

Compared with http://semantic-ui.com/kitchen-sink.html#/modules

![image](https://cloud.githubusercontent.com/assets/1164623/21141384/cd8f651c-c177-11e6-93fd-a461cf91fe8d.png)

tarvos21 replied:
Yes, I set the padding-bottom of last example to 35em, to keep the "transition" item in right rail become "active" and change color. To make it more comfortable, I may apply some more specific definitions on the last example.

xiaohanyu replied:
OK, but I still don't quite understand the connections between `35em` and "transition" item in right rail ...

I think `35em` is too large and unacceptable. Do me a favor and figure out a better solution. Many thanks.

tarvos21 replied:
OK, I will try my best.

xiaohanyu commented on content/themes/semantic-ui/amazon/index.slim:
```diff
@@ -9,8 +9,9 @@
       a.item data-tab="modules" Modules
 
-.ui.demo.container
+.ui.demo.container.basic.segment
```
Could you explain why would you like to make it a `.container.basic.segment`?

I checked http://semantic-ui.com/kitchen-sink.html page and found it is a `.basic.container`,

tarvos21 replied:
No, I can't definitely explain it now. If not added, the right rail will go to the top of the page.

xiaohanyu replied:
It's OK, I think maybe you should keep this question in your head and try to figure it out in your spare time.

xiaohanyu commented on content/static/stylesheets/default.sass:
```diff
@@ -130,6 +130,23 @@ img
+    width: 827px
+    .example:last-child:not(.position)
+      padding-bottom: 35em
+  &.basic.segment
```
Found `.basic.segment` here, 😄

tarvos21 replied:
Yes, making it a segment seems to make it easier to define the size of anything, at least for this page.

xiaohanyu commented:
Found a typo in your commit message: https://github.com/semantic-ui-forest/semantic-ui-forest/pull/2/commits/05aca509df8cacdc3b410dea8e4b110e39e0cef2

`screnn wider` -> `screen wider`

xiaohanyu commented on content/static/stylesheets/default.sass:
```diff
@@ -130,6 +130,23 @@ img
         color: inherit
 
+/* Position, width, margin modifications for right rail
+/* Only for screen wider than 1200px
```
I'm not quite sure whether I asked the right question. You write "Only for screen wider than 1200px". But if the code only works for "screen wider than 1200px", should we wrap it in a CSS media query?

Ref:
https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Using_media_queries

xiaohanyu commented on content/static/javascripts/docs.js:
```diff
@@ -0,0 +1,451 @@
+    $example             = $('.example'),
+
+   // alias
+    handler
```
lacks one space alignment for `// alias`

xiaohanyu commented:
Oh, the js code is really long, need some time to review this carefully.

xiaohanyu commented on content/static/javascripts/docs.js:
```diff
@@ -0,0 +1,451 @@
+      }
+    },
+
+      refreshSticky: function() {
```
Indentation error here.

xiaohanyu commented on content/static/javascripts/docs.js:
```diff
@@ -0,0 +1,451 @@
+      }
+    },
+
+   tryCreateMenu: function(event) {
```
Indentation error here.

tarvos21 committed:
Width and position for 1200px, 992px, 768px and smaller screens

- Define the width of tab and rail in different screens
- Make the width of the heading part same as that of the tab

xiaohanyu commented:
Wonderful job, thanks!
